src/components/HypothesisTesting/PerformTest.js

```diff
@@ -99,8 +99,23 @@ export default function PerformTest({ distType, shape, sides, mu0, equality, tes
       return jStat.ztest(zscore, sides)
     } else if (distType === 'T' && testType === 'oneSample') {
       return jStat.ttest(tscore, sampleSize - 1, sides)
+
     } else if (distType === 'Z' && testType !== 'oneSample') {
-      return jStat.ztest(zscoreTwoSample, sides);
+
+        if(zscoreTwoSample >= 0 && originalSampleMean > mu0) {  
```

Comment:
So I don't think we want to be comparing actual values here, rather we want to look at the language of the hypothesis itself. The `equality` prop passed to this component holds the comparison operator of the hypothesis - this will be useful here.